plugin/core/windows.py

```diff
@@ -557,6 +566,50 @@ def _check_window_closed(self) -> None:
         if not self._is_closing and not self._window.is_valid():
             self._handle_window_closed()
 
+    def _receive_progress_token(self, params: Dict[str, Any], client: Client, request_id: Any) -> None:
+        self._progress[params['token']] = dict()
+        client.send_response(Response(request_id, None))
+
+    def _show_progress(self, params: Dict[str, Any]) -> None:
+        token = params.get('token')
+        value = params.get('value')
+        if token and value:
+            if value.get('kind') == 'begin':
+                if token not in self._progress:
+                    self._progress[token] = dict()
+                self._progress[token]['title'] = value.get('title')  # mandatory
+                self._progress[token]['message'] = value.get('message')  # optional
+                status_msg = self._progress_string(token, value)
+                if status_msg:
+                    self._sublime.status_message(status_msg)
+            elif value.get('kind') == 'report':
+                status_msg = self._progress_string(token, value)
+                if status_msg:
+                    self._sublime.status_message(status_msg)
+            elif value.get('kind') == 'end':
```

Comment:
This is still overly cautious IMO. The `kind` **must** be present, or an editor does not know when to delete data associated with the token. If it's not present then a language server can arbitrarily increase the memory usage of an editor. Just use `value['kind']` and `value['title']`, `params['token']` and `params['value']`, it makes things easier to read.

Reply:
Ok.